starknet_devnet/blocks.py

```diff
@@ -89,52 +86,53 @@ def get_state_update(self, block_hash=None, block_number=None):
         return self.__state_updates.get(self.get_number_of_blocks() - 1) or self.origin.get_state_update()
 
     async def generate(
-        self, tx_wrapper: TransactionWrapper, state: StarknetState,
+        self, transaction: DevnetTransaction, state: StarknetState,
         state_root: bytes, state_update = None
-    ) -> Tuple[str, int]:
+    ) -> StarknetBlock:
         """
         Generates a block and stores it to blocks and hash2block. The block contains just the passed transaction.
         The `tx_wrapper.transaction` dict should contain a key `transaction`.
         Returns (block_hash, block_number).
         """
         block_number = self.get_number_of_blocks()
         timestamp = state.state.block_info.block_timestamp
-        signature = []
-        if "signature" in tx_wrapper.transaction["transaction"]:
-            signature = [int(sig_part, 16) for sig_part in tx_wrapper.transaction["transaction"]["signature"]]
+        signature = transaction.get_signature()
 
-        parent_block_hash = self.__get_last_block()["block_hash"] if block_number else fixed_length_hex(0)
-        sequencer_address = state.general_config.sequencer_address
-        gas_price = state.general_config.min_gas_price
+        if block_number == 0:
+            parent_block_hash = 0
+        else:
+            last_block: StarknetBlock = self.__get_last_block()
```

Comment:
Why declare last_block type explicitly, can't the return type of `__get_last_block` be used?

Reply:
[f7f7e48](https://github.com/Shard-Labs/starknet-devnet/pull/104/commits/f7f7e4821c8bb000b32f5d7d8904253b1ecfade2)